core/src/main/java/org/apache/iceberg/jdbc/JdbcViewOperations.java

```diff
@@ -0,0 +1,190 @@
+public class JdbcViewOperations extends BaseViewOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcViewOperations.class);
+  private final String catalogName;
+  private final TableIdentifier viewIdentifier;
+  private final FileIO fileIO;
+  private final JdbcClientPool connections;
+  private final Map<String, String> catalogProperties;
+
+  protected JdbcViewOperations(
+      JdbcClientPool dbConnPool,
+      FileIO fileIO,
+      String catalogName,
+      TableIdentifier viewIdentifier,
+      Map<String, String> catalogProperties) {
+    this.catalogName = catalogName;
+    this.viewIdentifier = viewIdentifier;
+    this.fileIO = fileIO;
+    this.connections = dbConnPool;
+    this.catalogProperties = catalogProperties;
+  }
+
+  @Override
+  protected void doRefresh() {
+    Map<String, String> view;
+
+    try {
+      view = JdbcUtil.getView(connections, catalogName, viewIdentifier);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e, "Interrupted during refresh");
+    } catch (SQLException e) {
+      // SQL exception happened when getting view from catalog
+      throw new UncheckedSQLException(
+          e, "Failed to get view %s from catalog %s", viewIdentifier, catalogName);
+    }
+
+    if (view.isEmpty()) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s", viewIdentifier);
+      } else {
+        this.disableRefresh();
+        return;
+      }
+    }
+
+    String newMetadataLocation = view.get(JdbcTableOperations.METADATA_LOCATION_PROP);
+    Preconditions.checkState(
+        newMetadataLocation != null, "Invalid view %s: metadata location is null", viewIdentifier);
+    refreshFromMetadataLocation(newMetadataLocation);
+  }
+
+  @Override
+  protected void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    boolean newView = base == null;
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+    try {
+      Map<String, String> view = JdbcUtil.getView(connections, catalogName, viewIdentifier);
+
+      if (base != null) {
+        validateMetadataLocation(view, base);
+        String oldMetadataLocation = base.metadataFileLocation();
+        // Start atomic update
+        LOG.debug("Committing existing view: {}", viewName());
+        updateView(newMetadataLocation, oldMetadataLocation);
+      } else {
+        // view not exists create it
+        LOG.debug("Committing new view: {}", viewName());
+        createView(newMetadataLocation);
+      }
+
+    } catch (SQLIntegrityConstraintViolationException e) {
+      if (currentMetadataLocation() == null) {
+        throw new AlreadyExistsException(e, "View already exists: %s", viewIdentifier);
+      } else {
+        throw new UncheckedSQLException(e, "View already exists: %s", viewIdentifier);
+      }
+
+    } catch (SQLException e) {
```

Comment:
`JdbcTableOperations` catches a few more exception, not sure if we need to do the same here?